### PR TITLE
add files with complete iteration exercises: Alex Kaudy

### PR DESCRIPTION
**Pull request by kaudyac**
Alex Kaudy (she/they) BE

### CODE REVIEW

File: lib/classroom.rb [hunk 1]
```diff
@@ -1,2 +1,25 @@
 class Classroom
-end
\ No newline at end of file
+  attr_reader :subject, :capacity, :students
```
Comment (mikedao) [inline lib/classroom.rb:2]: Split this up into individual lines

File: lib/classroom.rb [hunk 2]
```diff
@@ -1,2 +1,25 @@
...
+    @capacity = capacity
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student
+  end
+
+  def yell_at_students
+    students.each {|student| puts student.upcase} #.upcase is returning as an undefined method
```
Comment (mikedao) [inline lib/classroom.rb:15]: This method should return us the names uppercased, youre printing them out to the screen.

Comment (mikedao) [reply lib/classroom.rb:15]: Also default to the multi line enumerable syntax.

File: spec/classroom_spec.rb [hunk 3]
```diff
@@ -45,4 +45,34 @@
...
+  context 'capacity' do
+    it 'returns if the class is over capacity' do
+      classroom = Classroom.new('History', 4)
+      classroom.add_student('Mike')
+      classroom.add_student('Megan')
+      classroom.add_student('Bob')
+
+      expect(classroom.over_capacity?).to eq false
+    end
+  end
```
Comment (mikedao) [inline spec/classroom_spec.rb:60]: we're never really checking if it IS over capacity here

File: spec/classroom_spec.rb [hunk 4]
```diff
@@ -45,4 +45,34 @@
...
+      classroom.add_student('Mike')
+      classroom.add_student('Megan')
+      classroom.add_student('Bob')
+      classroom.add_student('James')
+      classroom.add_student('Cat')
+      classroom.add_student('Alice')
+
+      expect(classroom.students).to eq (["Bob", "James", "Cat", "Alice"])
+    end
+  end
```
Comment (mikedao) [inline spec/classroom_spec.rb:75]: This test does not kick out students

File: lib/classroom.rb [hunk 5]
```diff
@@ -1,2 +1,25 @@
...
+  end
+
+  def over_capacity?
+    students.size <= capacity
```
Comment (mikedao) [inline lib/classroom.rb:19]: Should be >=

**Review by mikedao — COMMENTED** [review]
Good start, make sure your tests are doing what youre testing for.

Score: Pass

File: lib/classroom.rb (hunk 5 above)
Comment (kaudyac) [reply lib/classroom.rb:19]: Noted and changed.

**Pull request closed without merging**